array_api_tests/test_linalg.py

```diff
@@ -901,6 +905,15 @@ def true_trace(x_stack, offset=0):
 
     _test_stacks(linalg.trace, x, **kw, res=res, dims=0, true_val=true_trace)
 
+
+def _conj(x):
+    """Work around xp.conj rejecting floats."""
+    if xp.isdtype(x.dtype, 'complex floating'):
```

Comment:
You won't be able to use this helper here because it won't be implemented in every library. I would suggest adding `is_complex_dtype` to dtype_helpers.py and using it here.

Reply:
Isn't it in the spec? https://data-apis.org/array-api/latest/API_specification/generated/array_api.isdtype.html

Reply:
If we used `isdtype` here, it would fail with plain PyTorch (for instance), not because `torch.linalg.vecdot` is wrong, but only because PyTorch doesn't implement `isdtype` yet. But that's unnecessary, because we can instead just test against complex dtypes here directly. 

In general in the test suite, we want to make the dependencies across tests as minimal as possible, especially in the case of functions that are definitely not yet implemented everywhere. Tests depending on other array API behavior is unavoidable, but pragmatism is the key point. We know that `dtype in [complex64, complex128]` will work in every library that we test and that `isdtype(dtype, 'complex floating')` will currently not.

Reply:
Ah OK, that's practicality, not purity. That makes total sense, thank you! Made the change.
(somewhat funny how it goes precisely the opposite to some review comments in scipy)

Reply:
Yes, that will probably the hardest thing to get used to, both here in the test suite and also in array-api-strict and array-api-compat. In the test suite, we have to be as minimal as possible with respect to the standard in order to accept as many things as possible. In strict and compat, we have to assume the libraries are what they are (instead of assuming they are array API packages), and wrap them to be more like the array API. In all instances, we consider specific implementation details about specific packages, even when those disagree with the standard.